src/Extensions/Diagnostics.HealthChecks/Internal/RestHealthCheckPublisher.cs

```diff
@@ -0,0 +1,104 @@
+		internal static string FabricNodeNameEnv = "Fabric_NodeName";
+
+		public RestHealthCheckPublisher(ILogger<ServiceFabricHealthCheckPublisher> logger,
+										RestHealthCheckPublisherOptions options,
+										ObjectPoolProvider objectPoolProvider) : base(objectPoolProvider)
+		{
+			m_clientWrapper = new(new Uri(options.RestHealthPublisherClusterEndpoint));
+			m_logger = logger;
+		}
+
+		public override async Task PublishAsync(HealthReport report, CancellationToken cancellationToken)
+		{
+			ServiceFabricHttpClient httpClient = m_clientWrapper.Get();
+			try
+			{
+				// We trust the framework to ensure that the report is not null and doesn't contain null entries.
+				foreach (KeyValuePair<string, HealthReportEntry> entryPair in report.Entries)
+				{
+					cancellationToken.ThrowIfCancellationRequested();
+					await PublishHealthInfoAsync(BuildSfHealthInformation(entryPair.Key, entryPair.Value), cancellationToken, httpClient);
+				}
+
+				cancellationToken.ThrowIfCancellationRequested();
+				await PublishHealthInfoAsync(BuildSfHealthInformation(report), cancellationToken, httpClient);
+			}
+			catch (Exception)
+			{
+				// Ignore, the service instance is closing.
+			}
+		}
+
+		internal static ServiceFabricCommon.HealthInformation FromSfHealthInformation(ServiceFabricHealth.HealthInformation healthInfo)
+		{
+			ServiceFabricCommon.HealthState healthState = healthInfo.HealthState switch
+			{
+				ServiceFabricHealth.HealthState.Invalid => ServiceFabricCommon.HealthState.Invalid,
+				ServiceFabricHealth.HealthState.Ok => ServiceFabricCommon.HealthState.Ok,
+				ServiceFabricHealth.HealthState.Warning => ServiceFabricCommon.HealthState.Warning,
+				ServiceFabricHealth.HealthState.Error => ServiceFabricCommon.HealthState.Error,
+				ServiceFabricHealth.HealthState.Unknown => ServiceFabricCommon.HealthState.Unknown,
+				_ => throw new ArgumentException($"'{healthInfo.HealthState}' is not a valid health status."),
+			};
+
+			ServiceFabricCommon.HealthInformation sfcHealthInfo = new(
+				   sourceId: healthInfo.SourceId,
+				   property: healthInfo.Property,
+				   healthState: healthState,
+				   timeToLiveInMilliSeconds: healthInfo.TimeToLive,
+				   description: healthInfo.Description,
+				   sequenceNumber: healthInfo.SequenceNumber.ToString(),
+				   removeWhenExpired: healthInfo.RemoveWhenExpired,
+				   healthReportId: healthInfo.HealthReportId
+		   );
+
+			return sfcHealthInfo;
+		}
+
+		internal async Task PublishHealthInfoAsync(ServiceFabricHealth.HealthInformation healthInfo, CancellationToken cancellationToken,
+			ServiceFabricHttpClient httpClient)
+		{
+			ServiceFabricCommon.HealthInformation sfcHealthInfo = FromSfHealthInformation(healthInfo);
+			string? nodeName = Environment.GetEnvironmentVariable(FabricNodeNameEnv);
+			if(nodeName == null)
+			{
+				m_logger.LogError(Tag.Create(), "Can't find node name.");
+				return;
+			}
```

Comment:
I think it would be better to get the node name once since it would never change.
An even better approach would be just to expose it as another field in `IExecutionContext` since we already getting it there. https://github.com/microsoft/Omex/blob/4732acfb475274e111504098d01332a15d0aa374/src/Extensions/Abstractions/ExecutionContext/BaseExecutionContext.cs#L63

Reply:
Ok, I assumed that they can dynamically rotate